src/components/Header/Header.tsx

```diff
@@ -5,12 +5,18 @@ import {
 	ThemeSwitchButton,
 } from "./Header.styles";
 
-export const Header = () => (
-	<StyledHeader>
-		<StyledH2>Where in the world?</StyledH2>
-		<ThemeSwitchButton type="button">
-			<StyledIcon className="fa-regular fa-moon"></StyledIcon>
-			Dark Mode
-		</ThemeSwitchButton>
-	</StyledHeader>
-);
+export interface HeaderProps {
+	toggleTheme: () => void;
+}
+
+export const Header = ({ toggleTheme }: HeaderProps) => {
```

Comment:
Why do you pass this as props? Use have a custom hook for that

Reply:
but It's not really important. You decide which solution to pick

Reply:
When I tried to use toggleTheme from custom hook directly inside Header component it did not work and did not change the state.
When I pass it as props (I saw in tutorials almost always they use props in similar situation) it works correct.